src/sumo.facade.ts

```diff
@@ -0,0 +1,347 @@
+
+// ===========================
+// Hash Functions
+// ===========================
+
+/**
+ * SHA-512 hash function
+ */
+export function crypto_hash_sha512(message: Uint8Array): Uint8Array {
+  return sha512(message);
+}
+
+/**
+ * BLAKE2b hash function (generic hash)
+ * Matches libsodium signature: crypto_generichash(outputLength, message, key?)
+ */
+export function crypto_generichash(
+  outputLength: number,
+  message: Uint8Array,
+  key: Uint8Array | null = null
+): Uint8Array {
+  if (key) {
+    return blake2b(message, { key, dkLen: outputLength });
+  }
+  return blake2b(message, { dkLen: outputLength });
+}
+
+// ===========================
+// Key Exchange Functions
+// ===========================
+
+/**
+ * Generate client session keys for key exchange
+ */
+export function crypto_kx_client_session_keys(
+  clientPub: Uint8Array,
+  clientPriv: Uint8Array,
+  serverPub: Uint8Array
+): CryptoKX {
+  // Perform ECDH
+  const sharedSecret = x25519.scalarMult(clientPriv, serverPub);
+
+  // Derive session keys using BLAKE2b
+  const sessionKeyMaterial = new Uint8Array(
+    clientPub.length + serverPub.length + sharedSecret.length
+  );
+  sessionKeyMaterial.set(clientPub, 0);
+  sessionKeyMaterial.set(serverPub, clientPub.length);
+  sessionKeyMaterial.set(sharedSecret, clientPub.length + serverPub.length);
+
+  // Generate rx and tx keys
+  const rxKey = blake2b(new Uint8Array([...sessionKeyMaterial, 0]), {
+    dkLen: 32,
+  });
+  const txKey = blake2b(new Uint8Array([...sessionKeyMaterial, 1]), {
+    dkLen: 32,
+  });
+
+  return {
+    sharedRx: rxKey,
+    sharedTx: txKey,
+  };
+}
+
+/**
+ * Generate server session keys for key exchange
+ */
+export function crypto_kx_server_session_keys(
+  serverPub: Uint8Array,
+  serverPriv: Uint8Array,
+  clientPub: Uint8Array
+): CryptoKX {
+  // Perform ECDH
+  const sharedSecret = x25519.scalarMult(serverPriv, clientPub);
+
+  // Derive session keys using BLAKE2b (note: swapped order compared to client)
+  const sessionKeyMaterial = new Uint8Array(
+    clientPub.length + serverPub.length + sharedSecret.length
+  );
+  sessionKeyMaterial.set(clientPub, 0);
+  sessionKeyMaterial.set(serverPub, clientPub.length);
+  sessionKeyMaterial.set(sharedSecret, clientPub.length + serverPub.length);
+
+  // Generate rx and tx keys (swapped compared to client)
+  const rxKey = blake2b(new Uint8Array([...sessionKeyMaterial, 1]), {
+    dkLen: 32,
+  });
+  const txKey = blake2b(new Uint8Array([...sessionKeyMaterial, 0]), {
+    dkLen: 32,
+  });
+
+  return {
+    sharedRx: rxKey,
+    sharedTx: txKey,
+  };
+}
+
+// ===========================
+// Symmetric Encryption (SecretBox)
+// ===========================
+
+/**
+ * Encrypt a message using ChaCha20Poly1305 (simplified implementation using Node.js crypto)
+ */
+export function crypto_secretbox_easy(
+  message: Uint8Array,
+  nonce: Uint8Array,
+  key: Uint8Array
+): Uint8Array {
+  // Using AES-256-GCM as a replacement for ChaCha20Poly1305
```

Comment:
@rbonestell we would like to keep chachapoly1305. This would also break any encrypted boxes someone already created using the lib. Is there a specific reason that i'm missing why AES is required here? thks!